File: backend/api/schemas/location.py
from pydantic import BaseModel, Field


class LocationCreate(BaseModel):
    title: str
    address: str
    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class LocationUpdate(BaseModel):
    title: str
    address: str
    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class LocationResponse(BaseModel):
    id: int
    title: str
    address: str
    latitude: float
    longitude: float
    position: int
    is_hidden: bool

File: frontend/src/types/location.ts
export interface LocationCreate {
  title: string
  address: string
  latitude: number
  longitude: number
  position?: number
  is_hidden?: boolean
}

export interface LocationUpdate {
  title: string
  address: string
  latitude: number
  longitude: number
  position?: number
  is_hidden?: boolean
}

export interface LocationResponse {
  id: number
  title: string
  address: string
  latitude: number
  longitude: number
  position: number
  is_hidden: boolean
}

export interface PublicScheduleResponse {
  id: number
  weekday: number
  start_time: string | null
  end_time: string | null
  is_day_off: boolean
}

export interface PublicLocationResponse {
  id: number
  title: string
  address: string
  latitude: number
  longitude: number
  position: number
  schedules: PublicScheduleResponse[]
}